src/modules/select/wallets/gnosis.ts

```diff
@@ -0,0 +1,54 @@
+import type SafeAppsSDK from '@gnosis.pm/safe-apps-sdk'
+import type { SafeInfo } from '@gnosis.pm/safe-apps-sdk'
+import { CommonWalletOptions, Helpers, WalletModule } from '../../../interfaces'
+import gnosisWalletIcon from '../wallet-icons/icon-gnosis'
+
+const getSafe = (sdk: SafeAppsSDK): Promise<SafeInfo | undefined> => {
+  return Promise.race([
+    sdk.getSafeInfo(),
+    // Timeout need as this method hangs until it can find the safe info
+    new Promise<undefined>(resolve => setTimeout(resolve, 200))
+  ])
+}
+
+function gnosis(options: CommonWalletOptions): WalletModule {
+  const { preferred, label, iconSrc, svg, networkId } = options
+
+  const network = networkId === 4 ? 'rinkeby.' : ''
+  const link = `https://${network}gnosis-safe.io/app`
+
+  return {
+    name: label || 'Gnosis Safe Wallet',
+    iconSrc,
+    svg: svg || gnosisWalletIcon,
+    wallet: async ({ createModernProviderInterface }: Helpers) => {
+      const sdk = new (await import('@gnosis.pm/safe-apps-sdk')).default()
+      const { SafeAppProvider } = await import('@gnosis.pm/safe-apps-provider')
+
+      const safe: SafeInfo | undefined = await getSafe(sdk)
+      // Checks if we are within the safe app context
+      if (!safe) return { provider: undefined, interface: null }
+
+      const provider = new SafeAppProvider(safe, sdk)
+      return {
+        provider,
+        interface: {
+          ...createModernProviderInterface(provider),
+          connect: () => Promise.resolve([safe.safeAddress])
```

Comment:
The SafeAppProvider doesn't implement the `eth_requestAccounts` rpc method so this is added to mimic it in the event that it gets called